smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy

```diff
@@ -430,142 +431,182 @@ def addDevice() {
 }
 
 def activity(dni,mode) {
-    def Params = [auth: state.HarmonyAccessToken]
-    def msg = "Command failed"
-    def url = ''
+    def tokenParam = [auth: state.HarmonyAccessToken]
+    def url
     if (dni == "all") {
-        url = "https://home.myharmony.com/cloudapi/activity/off?${toQueryString(Params)}"
+        url = "https://home.myharmony.com/cloudapi/activity/off?${toQueryString(tokenParam)}"
     } else {
         def aux = dni.split('-')
         def hubId = aux[1]
         if (mode == "hub" || (aux.size() <= 2) || (aux[2] == "off")){
-        	url = "https://home.myharmony.com/cloudapi/hub/${hubId}/activity/off?${toQueryString(Params)}"
+        	url = "https://home.myharmony.com/cloudapi/hub/${hubId}/activity/off?${toQueryString(tokenParam)}"
         } else {
-            def activityId = aux[2]
-        	url = "https://home.myharmony.com/cloudapi/hub/${hubId}/activity/${activityId}/${mode}?${toQueryString(Params)}"
+          def activityId = aux[2]
+        	url = "https://home.myharmony.com/cloudapi/hub/${hubId}/activity/${activityId}/${mode}?${toQueryString(tokenParam)}"
         }
 	}
-	try {
-    	httpPostJson(uri: url) { response ->
-        	if (response.data.code == 200 || dni == "all") {
-            	msg = "Command sent succesfully"
-            	state.aux = 0
-            } else {
-            	msg = "Command failed. Error: $response.data.code"
-            }
-		}
-    } catch (groovyx.net.http.HttpResponseException ex) {
-        log.error ex
-        if (state.aux == 0) {
-        	state.aux = 1
-        	activity(dni,mode)
-        } else {
-        	msg = ex
-            state.aux = 0
-        }
-    } catch(Exception ex) {
-    	msg = ex
-    }
-    runIn(10, "poll", [overwrite: true])
-    return msg
+  def params = [
+      uri: url,
+      contentType: 'application/json'
+  ]
+  asynchttp_v1.post('activityResponse', params, [:])
+  return "Command Sent"
+}
+
+def activityResponse(response, data) {
+	if (response.hasError()) {
+			log.error "Logitech Harmony - response has error: $response.errorMessage"
+      if (response.status == 401) { // token is expired
+          state.remove("HarmonyAccessToken")
+          log.warn "Logitech Harmony - Access token has expired"
+      }
+	} else {
+			def ResponseValues
+			try {
+					// json response already parsed into JSONElement object
+					ResponseValues = response.json
+			} catch (e) {
+					log.error "Logitech Harmony - error parsing json from response: $e"
+			}
+			if (ResponseValues) {
+          if (ResponseValues.code == 200) {
+              	log.trace "Command sent succesfully"
+              	poll()
```

Comment:
yes, so the activity device type and the harmony hub DTH gets updated with the correct value and status based on real data and not on assuming that the command was successful